components/script/dom/htmlcanvaselement.rs

```diff
@@ -202,16 +202,17 @@ impl HTMLCanvasElement {
 
         let data = if let Some(renderer) = self.ipc_renderer() {
             let (sender, receiver) = ipc::channel().unwrap();
-            let msg = CanvasMsg::FromLayout(FromLayoutMsg::SendData(sender));
+            let msg = CanvasMsg::FromScript(FromScriptMsg::SendPixels(sender));
             renderer.send(msg).unwrap();
 
             match receiver.recv().unwrap() {
-                CanvasData::Pixels(pixel_data)
-                    => pixel_data.image_data.to_vec(),
-                CanvasData::WebGL(_)
-                    // TODO(emilio): Not sure if WebGL canvas is required for 2d spec,
-                    // but I think it's not.
-                    => return None,
+                Some(pixels) => pixels,
+                None => {
+                    // TODO(emilio): Not sure if WebGL canvas is required for 2d
```

Comment:
Hmm, IIRC the last time I looked at this the canvas 2d spec only defined the behavior when another 2d canvas was provided, but yeah, worth investigating and fixing if required.